src/category_theory/single_obj.lean

```diff
@@ -64,46 +64,71 @@ protected def star : single_obj α := unit.star
-
-lemma to_End_def [monoid α] (x : α) : to_End x = x := rfl
+lemma to_End_def [monoid α] (x : α) : to_End α x = x := rfl
 
 /-- There is a 1-1 correspondence between monoid homomorphisms `α → β` and functors between the
     corresponding single-object categories. It means that `single_obj` is a fully faithful
     functor. -/
-def map_hom_equiv (α : Type u) (β : Type v) [monoid α] [monoid β] :
-  { f : α → β // is_monoid_hom f } ≃ (single_obj α) ⥤ (single_obj β) :=
+def map_hom (α : Type u) (β : Type v) [monoid α] [monoid β] :
+  (α →* β) ≃ (single_obj α) ⥤ (single_obj β) :=
 { to_fun := λ f,
   { obj := id,
-    map := λ _ _, f.1,
-    map_id' := λ _, f.2.map_one,
-    map_comp' := λ _ _ _ x y, @is_mul_hom.map_mul _ _ _ _ _ f.2.1 y x },
-  inv_fun := λ f, ⟨@functor.map _ _ _ _ f (single_obj.star α) (single_obj.star α),
-    { map_mul := λ x y, f.map_comp y x, map_one := f.map_id _ }⟩,
-  left_inv := λ ⟨f, hf⟩, rfl,
-  right_inv := assume f, by rcases f; obviously }
+    map := λ _ _, ⇑f,
+    map_id' := λ _, f.map_one,
+    map_comp' := λ _ _ _ x y, f.map_mul y x },
+  inv_fun := λ f,
+    { to_fun := @functor.map _ _ _ _ f (single_obj.star α) (single_obj.star α),
+      map_one' := f.map_id _,
+      map_mul' := λ x y, f.map_comp y x },
+  left_inv := λ ⟨f, h₁, h₂⟩, rfl,
+  right_inv := λ f, by cases f; obviously }
+
+lemma map_hom_id (α : Type u) [monoid α] : map_hom α α (monoid_hom.id α) = 𝟭 _ := rfl
+
+lemma map_hom_comp {α : Type u} {β : Type v} [monoid α] [monoid β] (f : α →* β)
+  {γ : Type w} [monoid γ] (g : β →* γ) :
+  map_hom α γ (g.comp f) = map_hom α β f ⋙ map_hom β γ g :=
+rfl
+
+end single_obj
+
+end category_theory
+
+open category_theory
+
+namespace monoid_hom
 
 /-- Reinterpret a monoid homomorphism `f : α → β` as a functor `(single_obj α) ⥤ (single_obj β)`.
-See also `map_hom_equiv` for an equivalence between these types. -/
-@[reducible] def map_hom {α : Type u} {β : Type v} [monoid α] [monoid β]
-  (f : α → β) [hf : is_monoid_hom f] :
+See also `category_theory.single_obj.map_hom` for an equivalence between these types. -/
+@[reducible] def to_functor {α : Type u} {β : Type v} [monoid α] [monoid β] (f : α →* β) :
   (single_obj α) ⥤ (single_obj β) :=
-map_hom_equiv α β ⟨f, hf⟩
-
-lemma map_hom_id {α : Type u} [monoid α] : map_hom (@id α) = 𝟭 _ := rfl
+single_obj.map_hom α β f
 
-lemma map_hom_comp {α : Type u} {β : Type v} [monoid α] [monoid β] (f : α → β) [is_monoid_hom f]
-  {γ : Type w} [monoid γ] (g : β → γ) [is_monoid_hom g] :
-  map_hom f ⋙ map_hom g = map_hom (g ∘ f) :=
+@[simp] lemma id_to_functor (α : Type u) [monoid α] : (id α).to_functor = 𝟭 _ := rfl
+@[simp] lemma comp_to_functor {α : Type u} {β : Type v} [monoid α] [monoid β] (f : α →* β)
+  {γ : Type w} [monoid γ] (g : β →* γ) :
+  (g.comp f).to_functor = f.to_functor ⋙ g.to_functor :=
 rfl
 
-end single_obj
+end monoid_hom
 
-end category_theory
+namespace units
+
+variables (α : Type u) [monoid α]
+
+def to_Aut : units α ≃* Aut (single_obj.star α) :=
```

Comment:
If you decide to change the direction of `to_End` (see above), then this one should change as well.

Reply:
Let's leave this as is (see above).